packages/xo-server/src/xo-mixins/proxies.mjs

```diff
@@ -165,14 +181,18 @@ export default class Proxy {
 
   async getProxy(id) {
     const proxy = await this._getProxy(id)
-    addProxyUrl.call(this, proxy)
+    await populateProxy.call(this, proxy)
 
     return proxy
   }
 
   async getAllProxies() {
     const proxies = await this._db.get()
-    proxies.forEach(addProxyUrl, this)
+    await Promise.all(
+      proxies.map(async proxy => {
+        await populateProxy.call(this, proxy)
+      })
+    )
```

Comment:
returning a promise and await the result has the same  behavior  as awaiting the promise in an async method, because the async method is internally converted to a promise, that is awaited